Readability.js

```diff
@@ -346,8 +346,9 @@ Readability.prototype = {
     }
 
     curTitle = curTitle.trim();
-
-    if (curTitle.split(' ').length <= 4)
+    var curTitleLen = curTitle.split(' ').length;
+    if (curTitleLen <=4 && ( !origTitle.match(/ [\\\/>»] /)
+        || curTitleLen != origTitle.replace(/[\|\-\\\/>» ]+/g, " ").split(' ').length -1))
```

Comment:
Certainly. So normally we check in line 350 if our new title got shorter than 4 words and if yes, use the origTitle instead but this only makes sense if the current title isn't a hierarchical one because here we don't have a problem with the title having less than 4 words because it probably means we correctly found the last layer of the hierarchy. 

Of course, this approach is way too broad and would fail some tests so I limited it to only removing the root of the hierarchy.

That's the second part of the if, it boils down to: hierarchical title implies: the new title should only be one word shorter than the original title.

Point of interes is now line 317-321:
```Javascript
if (curTitle.match(/ [\|\-\\\/>»] /)) {
  curTitle = origTitle.replace(/(.*)[\|\-\\\/>»] .*/gi, '$1');

  if (curTitle.split(' ').length < 3)
  curTitle = origTitle.replace(/[^\|\-\\\/>»]*[\|\-\\\/>»](.*)/gi, '$1');
}
```
because in combination it is possible that we now only shorten hierarchical titles with a maximum of two layers which would basically defeat the whole point of all this. But on the other side, this means less interference with the existing behavior. My test use case only contains titles with two layers and I am not sure if there are any with more layers in the wild that would qualify for this kind shortening.

@Your eslint config: no problem, but you could add a line to the test section of the readme to make first-time users aware of eslint (I only found out when my pull request failed the  continuous-integration check)